plugins/trace_api_plugin/include/eosio/trace_api/chain_extraction.hpp

```diff
@@ -100,10 +101,20 @@ class chain_extraction_impl_type {
             if( it != cached_traces.end() ) {
                traces.emplace_back( to_transaction_trace<transaction_trace_t>( it->second ));
             }
+            tt.ids.insert(id);
          }
          clear_caches();
 
+         //store transaction ids
+         if (tt.ids.size() == 0){
+            transaction_id_type  place_holder_id{"0"};
```

Comment:
When a block has no transaction, a new entry with a meaningless id for the block will still be added into a trx id log as a place holder.  So id slices have consistent block numbers with trace slices. It will make deleting slices during log maintenance easier. 

If this is an acceptable solution, I should change "0" to some unique id because I have just changed the container for the ids from std::vector to std::set.

Reply:
Why can't the `ids` `set` just be empty?

Reply:
Set the ids be empty when a block has no transaction. Thanks.